### ISSUE TITLE
Call to undefined method Sensor::unit()

### ISSUE THREAD

**Issue by lucasRolff** [librenms/librenms#16832]
### The problem

On a newly installed LibreNMS, the overview pages of devices result in images giving 500 internal server error.

This is across devices such as Arista switches, APC, Raritan and RNX.ch PDUs:
![https://snaps.hcdn.dk/65XP5Sl76TzFy40UrZebhviheTmRC0mByQyZ96e1rVcKJ.png](https://snaps.hcdn.dk/65XP5Sl76TzFy40UrZebhviheTmRC0mByQyZ96e1rVcKJ.png)

Per the /opt/librenms/logs/ file I see:

```
Call to undefined method App\Models\Sensor::unit() {"userId":1,"exception":"[object] (BadMethodCallException(code: 0): Call to undefined method App\\Models\\Sensor::unit() at /opt/librenms/vendor/laravel/framework/src/Illuminate/Support/Traits/ForwardsCalls.php:67)"}
Call to undefined method App\Models\Sensor::unit() {"userId":1,"exception":"[object] (BadMethodCallException(code: 0): Call to undefined method App\\Models\\Sensor::unit() at /opt/librenms/vendor/laravel/framework/src/Illuminate/Support/Traits/ForwardsCalls.php:67)"}
Call to undefined method App\Models\Sensor::unit() {"userId":1,"exception":"[object] (BadMethodCallException(code: 0): Call to undefined method App\\Models\\Sensor::unit() at /opt/librenms/vendor/laravel/framework/src/Illuminate/Support/Traits/ForwardsCalls.php:67)"}
Call to undefined method App\Models\Sensor::unit() {"userId":1,"exception":"[object] (BadMethodCallException(code: 0): Call to undefined method App\\Models\\Sensor::unit() at /opt/librenms/vendor/laravel/framework/src/Illuminate/Support/Traits/ForwardsCalls.php:67)"}
```

### Output of ./validate.php

```txt
===========================================
Component | Version
--------- | -------
LibreNMS  | 24.11.0-73-gf170d9a5d (2024-12-07T00:11:00+00:00)
DB Schema | 2024_11_22_135845_alert_log_refactor_indexes (310)
PHP       | 8.3.6
Python    | 3.12.3
Database  | MariaDB 10.11.8-MariaDB-0ubuntu0.24.04.1
RRDTool   | 1.7.2
SNMP      | 5.9.4.pre2
===========================================

[OK]    Composer Version: 2.8.3
[OK]    Dependencies up-to-date.
[OK]    Database connection successful
[OK]    Database connection successful
[OK]    Database Schema is current
[OK]    SQL Server meets minimum requirements
[OK]    lower_case_table_names is enabled
[OK]    MySQL engine is optimal
[OK]    Database and column collations are correct
[OK]    Database schema correct
[OK]    MySQL and PHP time match
[OK]    Active pollers found
[OK]    Dispatcher Service not detected
[OK]    Locks are functional
[OK]    Python poller wrapper is polling
[OK]    Redis is unavailable
[OK]    rrd_dir is writable
[OK]    rrdtool version ok
```

### Anything in the logs that might be useful for us?

```txt
Call to undefined method App\Models\Sensor::unit() {"userId":1,"exception":"[object] (BadMethodCallException(code: 0): Call to undefined method App\\Models\\Sensor::unit() at /opt/librenms/vendor/laravel/framework/src/Illuminate/Support/Traits/ForwardsCalls.php:67)"}
Call to undefined method App\Models\Sensor::unit() {"userId":1,"exception":"[object] (BadMethodCallException(code: 0): Call to undefined method App\\Models\\Sensor::unit() at /opt/librenms/vendor/laravel/framework/src/Illuminate/Support/Traits/ForwardsCalls.php:67)"}
Call to undefined method App\Models\Sensor::unit() {"userId":1,"exception":"[object] (BadMethodCallException(code: 0): Call to undefined method App\\Models\\Sensor::unit() at /opt/librenms/vendor/laravel/framework/src/Illuminate/Support/Traits/ForwardsCalls.php:67)"}
Call to undefined method App\Models\Sensor::unit() {"userId":1,"exception":"[object] (BadMethodCallException(code: 0): Call to undefined method App\\Models\\Sensor::unit() at /opt/librenms/vendor/laravel/framework/src/Illuminate/Support/Traits/ForwardsCalls.php:67)"}
```

**Comment by pandalion98** [comment]
I've created a pull request that fixes this issue (#16837). I'm also facing the same issue.

**Comment by laf** [comment]
Already fixed, please update your install
